packages/profile-security-nodejs/src/implementation.ts

```diff
@@ -0,0 +1,573 @@
+  }
+
+  static getInstanceOrThrow(): Implementation {
+    if (!Implementation.instance) {
+      throw new Error("Implementation instance not found. Make sure you have initialized the plugin.");
+    }
+
+    return Implementation.instance;
+  }
+
+  static reset(): void {
+    Implementation.instance = undefined;
+  }
+
+  constructor(protected pluginConfig: SuperTokensPluginProfileSecurityNormalisedConfig) {}
+
+  getConfigForClient = async function (
+    this: Implementation,
+    // props needed for overriding
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    props: { userContext: any; session: SessionContainerInterface },
+  ): Promise<
+    { status: "OK"; config: SuperTokensPluginProfileSecurityNormalisedConfig } | { status: "ERROR"; message: string }
+  > {
+    return {
+      status: "OK",
+      config: {
+        enableSettingPassword: this.pluginConfig.enableSettingPassword,
+        enableThirdPartyLinkning: this.pluginConfig.enableThirdPartyLinkning,
+        enableMfaConfiguration: this.pluginConfig.enableMfaConfiguration,
+      },
+    };
+  };
+
+  setUserPassword = async function (
+    this: Implementation,
+    {
+      userId,
+      email,
+      password,
+      session,
+      userContext,
+    }: {
+      userId: string;
+      email: string;
+      password: string;
+      session: SessionContainerInterface;
+      userContext: any;
+    },
+  ): Promise<{ status: "OK" } | { status: "ERROR"; message: string }> {
+    const user = await getUser(userId, userContext);
+    if (!user) {
+      return { status: "ERROR", message: "User not found" };
+    }
+
+    const passwordLoginMethods = user.loginMethods.filter((lm) => lm.recipeId === "emailpassword");
+    if (passwordLoginMethods.length) {
+      return {
+        status: "ERROR",
+        message: "User already has a password set. Please use the change password feature to update your password.",
+      };
+    }
+
+    const verifiedLoginMethod = user.loginMethods.find((lm) => lm.email === email && lm.verified);
+    if (!verifiedLoginMethod) {
+      return {
+        status: "ERROR",
+        message: "The user does not have this email address or is not verified",
+      };
+    }
+
+    const signUpResult = await signUp(session!.getTenantId(), email, password, session!, userContext);
+    if (signUpResult.status === "EMAIL_ALREADY_EXISTS_ERROR") {
+      return {
+        status: "ERROR",
+        message:
+          "There already exists a user with this email address. Please use the change password feature to update your password.",
+      };
+    }
+    if (signUpResult.status === "LINKING_TO_SESSION_USER_FAILED") {
+      return {
+        status: "ERROR",
+        message: "Could not link the new password to the user. Please contact support.",
+      };
+    }
+    if (signUpResult.status !== "OK") {
+      return {
+        status: "ERROR",
+        message: "Password setting failed. Please contact support.",
```

Comment:
If the return value from signup has an error code, we may want to communicate that here.